### PR TITLE
Enable theme overrides

### PR DESCRIPTION
**Pull request by dominik-zeglen**
This PR adds new pages to storybook and ability to override MUI theme from macaw's perspective.

Screenshot from 2021-07-06 16-55-02
Screenshot from 2021-07-06 16-55-17

### CODE REVIEW

File: src/theme/ThemeProvider.tsx [hunk 1]
```diff
@@ -1,4 +1,6 @@
+import type { Theme } from "@material-ui/core/styles";
 import { MuiThemeProvider } from "@material-ui/core/styles";
+import { merge } from "lodash";
```
Comment (orzechdev) [inline src/theme/ThemeProvider.tsx:3]: Isn't it better to change it to?:
```suggestion
import merge from "lodash/merge";
```

Comment (dominik-zeglen) [reply src/theme/ThemeProvider.tsx:3]: `tsdx` has babel plugin for this, so no need to use deep imports

**Review by kamilpastuszka — APPROVED** [review]
Changes look good. A README file could be added to the project.

File: stories/index.stories.tsx [hunk 2]
```diff
@@ -1,16 +1,160 @@
...
+        dashboard interface and use it to create designs for your specific
+        purposes. Use the left-side navigation to choose between different pages
+        on which you can find categorized UI Components.
+      </Typography>
+      <Typography className={classes.paragraph} component="p">
+        This project is in 'view only' mode, which means that you will have to
+        duplicate it to make any changes and create your own solutions.
+      </Typography>
+      <Typography className={classes.paragraph} component="p">
+        Have a great time working on your designs and empowering your users
```
Comment (jwm0) [inline stories/index.stories.tsx:66]:
```suggestion
        Have a great time working on your designs and empowering your users.
```

**Review by jwm0 — CHANGES_REQUESTED** [review]

**Review by jwm0 — APPROVED** [review]

File: stories/index.stories.tsx [hunk 3]
```diff
@@ -1,16 +1,160 @@
...
     <div>
-      Macaw UI
-      <Button color="primary" variant="contained">
-        Go to components
-      </Button>
+      <SVG className={classes.logo} src={logo} />
+      <Typography className={classes.paragraph} component="p">
+        Welcome to the MacawUI.
+        <br />
+        This is an official UI Design Kit for Saleor, a GraphQL-First e-commerce
```
Comment (kamilpastuszka) [inline stories/index.stories.tsx:52]:
```suggestion
        This is an official UI Design Kit for Saleor - a GraphQL-First e-commerce
```

File: stories/index.stories.tsx [hunk 4]
```diff
@@ -1,16 +1,160 @@
...
+        platform for perfectionists.
+      </Typography>
+      <Typography className={classes.paragraph} component="p">
+        You can find most of the elements used in the creation of Saleor’s
+        dashboard interface and use it to create designs for your specific
+        purposes. Use the left-side navigation to choose between different pages
```
Comment (kamilpastuszka) [inline stories/index.stories.tsx:58]:
```suggestion
        purposes. Use the left-hand side navigation to choose between different pages
```

**Pull request merged**